tools/pdffontsinfo.py

```diff
@@ -0,0 +1,222 @@
+            for page in PDFPage.get_pages(
+                fp,
+                pagenos,
+                maxpages=maxpages,
+                password=password,
+                caching=caching,
+                check_extractable=True,
+            ):
+                page.rotate = (page.rotate + rotation) % 360
+                interpreter.process_page(page)
+
+        l_font_names = []
+        l_font_types = []
+        l_font_enc = []
+        l_font_uni = []
+
+        # Collecting data from sources
+        for fontkey in rsrcmgr._cached_fonts:
+            # Font = class (PDFType1Font etc)
+            font = rsrcmgr._cached_fonts[fontkey]
+            # Type = TrueType/Type 1/Type 3/Type CID
+            font_type = font.__class__.__name__
+            # Name = Helvetica etc
+            font_name = font.basefont
+
+            font_enc = font.get_encoding_name()
+
+            font_uni = font.get_toUnicode()
+
+            l_font_types.append(font_type)
+            l_font_names.append(font_name)
+            l_font_enc.append(font_enc)
+            l_font_uni.append(font_uni)
+
+        fonts = {}
+        fonts['names'] = l_font_names
+        fonts['types'] = l_font_types
+        fonts['enc'] = l_font_enc
+        fonts['uni'] = l_font_uni
+
+        device.close()
+        outfp.write(fname+" fonts:\n")
+        outfp.write(fonts2txt(fonts))
+        outfp.write('\n')
+
+        # delete the cache
+        rsrcmgr._cached_fonts = {}
+
+    if outfile:
+        outfp.close()
+
+
+def fonts2txt(fonts):
+    ''' Transforms the font information in a nice text table following
+        this pattern:
+
+        Name                       Type     Encoding             Uni
+        -------------------------- -------- -------------------- ---
+        Times-Roman                Type 1   Custom               no
+        Times-Bold                 Type 1   Standard             no
+        Helvetica                  Type 1   Custom               no
+        Helvetica-Bold             Type 1   Standard             no
+
+        The table header: 4 titles: 'Name', 'Type', 'Encoding', 'Uni'.
+        The width of the columns in chars is the following: 25 for Name,
+            10 for Type, 20 for Encoding & 3 for Uni.
+        Between each column there is a separating whitespace.
+        The header is separated from the data with a sequence of dashes
+            ('-') as the width of each column.
+
+        Params:
+            - fonts : dictionary with 4 lists. The keys are the following:
+                'names', 'types', 'enc', 'uni'
+                All lists have the same number of items
+                - names: list of strings with the names of the fonts
+                - types: list of strings with the names of the types
+                - enc: list of strings with the names of the encoding
+                    types
+                - uni: list of booleans that shows if it is unicode or not
+
+    '''
+    def process_line(list_info):
```

Comment:
I think the code becomes easier to use if you use one argument for each value instead of an array
```suggestion
    def process_line(name, type, encoding, unicode):
```

Reply:
Great suggestion! Changes implemented :D